File: app/src/main/java/com/bmco/cratesiounofficial/models/Links.java

package com.bmco.cratesiounofficial.models;

import java.util.HashMap;
import java.util.Map;
import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
    "owners",
    "reverse_dependencies",
    "version_downloads",
    "versions"
})
public class Links {

    @JsonProperty("owners")
    private String owners;
    @JsonProperty("reverse_dependencies")
    private String reverseDependencies;
    @JsonProperty("version_downloads")
    private String versionDownloads;
    @JsonProperty("versions")
    private String versions;
    @JsonIgnore
    private Map<String, Object> additionalProperties = new HashMap<String, Object>();

    @JsonProperty("owners")
    public String getOwners() {
        return owners;
    }

    @JsonProperty("owners")
    public void setOwners(String owners) {
        this.owners = owners;
    }

    @JsonProperty("reverse_dependencies")
    public String getReverseDependencies() {
        return reverseDependencies;
    }

    @JsonProperty("reverse_dependencies")
    public void setReverseDependencies(String reverseDependencies) {
        this.reverseDependencies = reverseDependencies;
    }

    @JsonProperty("version_downloads")
    public String getVersionDownloads() {
        return versionDownloads;
    }

    @JsonProperty("version_downloads")
    public void setVersionDownloads(String versionDownloads) {
        this.versionDownloads = versionDownloads;
    }

    @JsonProperty("versions")
    public String getVersions() {
        return versions;
    }

    @JsonProperty("versions")
    public void setVersions(String versions) {
        this.versions = versions;
    }

    @JsonAnyGetter
    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    @JsonAnySetter
    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }

}

File: app/src/main/java/com/bmco/cratesiounofficial/models/Links.kt
package com.bmco.cratesiounofficial.models

import java.io.Serializable
import java.util.HashMap
import com.fasterxml.jackson.annotation.JsonAnyGetter
import com.fasterxml.jackson.annotation.JsonAnySetter
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonPropertyOrder

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder("owners", "reverse_dependencies", "version_downloads", "versions")
class Links : Serializable {

    @JsonProperty("owners")
    @get:JsonProperty("owners")
    @set:JsonProperty("owners")
    var owners: String? = null
    @JsonProperty("reverse_dependencies")
    @get:JsonProperty("reverse_dependencies")
    @set:JsonProperty("reverse_dependencies")
    var reverseDependencies: String? = null
    @JsonProperty("version_downloads")
    @get:JsonProperty("version_downloads")
    @set:JsonProperty("version_downloads")
    var versionDownloads: String? = null
    @JsonProperty("versions")
    @get:JsonProperty("versions")
    @set:JsonProperty("versions")
    var versions: String? = null
    @JsonIgnore
    private val additionalProperties = HashMap<String, Any>()

    @JsonAnyGetter
    fun getAdditionalProperties(): Map<String, Any> {
        return this.additionalProperties
    }

    @JsonAnySetter
    fun setAdditionalProperty(name: String, value: Any) {
        this.additionalProperties[name] = value
    }

}
